### PR TITLE
Small improvements in scripting

### PR DESCRIPTION
**Pull request by castelblanque**
Signed-off-by: Rafa Castelblanque <rcastelblanq@vmware.com>

### Description of the change

Added some improvements (e.g. variable checks) to a E2E script and the Pinniped makefile of the project.

### Benefits

More robust scripts in the project, and usable from local command line too.

### Possible drawbacks

N/A

### CODE REVIEW

**Review by beni0888 — APPROVED** [review]
Looks good!

File: script/makefiles/deploy-dev-for-pinniped.mk [hunk 1]
```diff
@@ -50,6 +52,7 @@ delete-pinniped-jwt-authenticator:
 	kubectl --kubeconfig=${CLUSTER_CONFIG_FOR_PINNIPED} delete -f ./site/content/docs/latest/reference/manifests/kubeapps-pinniped-jwt-authenticator.yaml
 	kubectl --kubeconfig=${ADDITIONAL_CLUSTER_CONFIG_FOR_PINNIPED} delete -f ./site/content/docs/latest/reference/manifests/kubeapps-pinniped-jwt-authenticator.yaml
 
+# Using old version for deletion due to error for missing secret "pinniped-concierge-impersonation-proxy" not found
```
Comment (antgamdia) [inline script/makefiles/deploy-dev-for-pinniped.mk:55]: Mmm, that's weird. Seems like an issue to report to our Pinniped friends, doesn't it?

**Review by antgamdia — APPROVED** [review]
Nice, thanks!

File: script/makefiles/deploy-dev-for-pinniped.mk (hunk 1 above)
Comment (castelblanque) [reply script/makefiles/deploy-dev-for-pinniped.mk:55]: Not sure if that is a capability that they want to allow, but created issue [here](https://github.com/vmware-tanzu/pinniped/issues/1251).

Comment (cfryanr) [reply script/makefiles/deploy-dev-for-pinniped.mk:55]: Thanks for reporting this! Please see https://github.com/vmware-tanzu/pinniped/issues/1251#issuecomment-1210935999.

**Pull request merged**